lpc55_sign/src/signed_image.rs

```diff
@@ -360,16 +364,129 @@ pub fn remove_image_signature(mut img: Vec<u8>) -> Result<Vec<u8>, Error> {
     }
 
     // Plain images have a length of 0
-    LittleEndian::write_u32(&mut img[0x20..0x24], 0);
+    LittleEndian::write_u32(&mut img[HEADER_IMAGE_LENGTH], 0);
 
     // Set imageType to a plain image
-    LittleEndian::write_u32(&mut img[0x24..0x28], BootImageType::PlainImage as u32);
+    LittleEndian::write_u32(
+        &mut img[HEADER_IMAGE_TYPE],
+        BootImageType::PlainImage as u32,
+    );
 
     // Clear the offset to the certificate header
-    LittleEndian::write_u32(&mut img[0x28..0x2c], 0);
+    LittleEndian::write_u32(&mut img[HEADER_OFFSET], 0);
 
     // Strip the certificate table
     img.resize(cert_table_offset as usize, 0u8);
 
     Ok(img)
 }
+
+/// Computed from an LPC55 signed image: the certificate header, X.509
+/// certificates, and the root key table & its hash. See UM11126 §7.35.
+pub struct CertBlock {
+    /// Certificate header; see UM11126 Fig 18.
+    pub header: CertHeader,
+
+    /// X.509 certificates in root-to-signing-key order.
+    pub certs: Vec<Certificate>,
+
+    /// Root Key Table: SHA2-256 hashes of the four public
+    /// root keys that may be used to validate this image.
+    pub root_key_table: [[u8; 32]; 4],
+
+    /// Root Key Table Hash (RKTH): SHA2-256 hash of the root key table.
+    /// Uniquely identifies the set of keys that might validate this image;
+    /// may be matched against the RKTH field in a CMPA page.
+    pub root_key_table_hash: [u8; 32],
+}
+
+/// Given an LPC55 signed image, decode the certificate block and return
+/// it, a hash of the image contents (not including the signature), and the
+/// purported signature of that hash. Does *not* validate the signature or
+/// the certificates, only checks that they decode ok.
+pub fn image_certs_and_sig(image: &[u8]) -> Result<(CertBlock, Sha256, Signature), Error> {
+    // Check the image length.
+    let image_len = u32::from_le_bytes(image[HEADER_IMAGE_LENGTH].try_into()?);
+    if (image_len as usize) > image.len() {
```

Comment:
Are there cases where `image_len < image.len()` is valid, or should we check for equality here?